aws/src/main/scala/com/gu/contentapi/client/IAMSigner.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.contentapi.client
+
+import com.amazonaws.DefaultRequest
+import com.amazonaws.auth.{AWS4Signer, AWSCredentials}
+import com.amazonaws.http.HttpMethodName
+import collection.JavaConverters._
+
+object IAMSigner {
+  /**
+    * For api-gateway authorization.
+    * Returns the given set of headers, updated to include AWS sig4v signed headers based on the request and the credentials
+    */
+  def addIAMHeaders(headers: Map[String, String], url: String, credentials: AWSCredentials, awsRegion: String): Map[String, String] = {
```

Comment:
is there not a need for signing an existing request?

Reply:
Different apps use different http clients, so it's up to the calling code to add the returned headers